odl/tomo/geometry/fanbeam.py

```diff
@@ -0,0 +1,327 @@
+        det_rad = float(det_rad)
+        if det_rad <= 0:
+            raise ValueError('detector circle radius {} is not positive.'
+                             ''.format(det_rad))
+
+        if agrid is not None:
+            if not isinstance(agrid, TensorGrid):
+                raise TypeError('angle grid {!r} is not a `TensorGrid` '
+                                'instance.'.format(agrid))
+            if not angle_intvl.contains_set(agrid):
+                raise ValueError('angular grid {} not contained in angle '
+                                 'interval {}.'.format(agrid, angle_intvl))
+
+        super().__init__()
+        self._motion_params = angle_intvl
+        self._src_radius = src_rad
+        self._det_radius = det_rad
+        self._motion_grid = agrid
+        self._motion_params_offset = angle_offset
+
+    @property
+    def motion_params(self):
+        """Motion parameters of this geometry."""
+        return self._motion_params
+
+    @property
+    def motion_params_offset(self):
+        """Offset to motion parameters. """
+        return self._motion_params_offset
+
+    @property
+    def motion_grid(self):
+        """Sampling grid for this geometry's motion parameters."""
+        return self._motion_grid
+
+    @property
+    def angle_intvl(self):
+        """Angles (= motion parameters) of this geometry."""
+        return self._motion_params
+
+    @property
+    def angle_grid(self):
+        """Angle (= motion parameter) sampling grid of this geometry."""
+        return self._motion_grid
+
+    @property
+    def angle_offset(self):
+        """Offset to the rotation angle in the azimuthal plane.
+
+        The actual angles then reside within `angle_offset` + `angle_intvl`.
+        """
+        return self._motion_params_offset
+
+    @property
+    def src_radius(self):
+        """Source circle radius of this geometry."""
+        return self._src_radius
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
+
+    @property
+    def ndim(self):
+        """Number of dimensions of this geometry."""
+        return 2
+
+    def det_refpoint(self, angle):
+        """The detector reference point function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+
+            The reference point on the circle with radius :math:`R` at a given
+            rotation angle :math:`\\phi`, defined as :math:`R(\\cos\\phi,
+            \\sin\\phi)`
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+        return self.det_radius * np.array([np.cos(angle), np.sin(angle)])
```

Comment:
What about `angle_offset`? I have this feeling (once more) that `angle_offset` will be a prime source of errors.